src/utils/image.js

```diff
@@ -304,6 +307,95 @@ export class RawImage {
         return this._update(newData, this.width, this.height, 4);
     }
 
+    /**
+     * Get the pixel at the given coordinates.
+     * @param {number} x The x coordinate of the pixel.
+     * @param {number} y The y coordinate of the pixel.
+     * @returns {Promise<DataArray>} The pixel value. The length of the array will match the channels.
+     */
+    async getPixel(x, y) {
+        x = Math.max(Math.min(x, this.width), 0);
+        y = Math.max(Math.min(y, this.height), 0);
+
+        // Calculate base index, taking into account number of channels.
+        const baseIndex = ((y * this.width) + x) * this.channels;
+        return this.data.slice(baseIndex, baseIndex + this.channels);
+    }
+
+    /**
+     * Set the pixel at the given coordinates.
+     * @param {number} x The x coordinate of the pixel.
+     * @param {number} y The y coordinate of the pixel.
+     * @param {DataArray} value The pixel value. The length of the array should match the channels.
+     * @returns {Promise<RawImage>}
+     * @throws {Error} If the number of channels in the value does not match the number of channels in the image.
+     */
+    async setPixel(x, y, value) {
+        x = Math.max(Math.min(x, this.width), 0);
+        y = Math.max(Math.min(y, this.height), 0);
+
+        if (value.length !== this.channels) {
+            throw new Error(`Expected ${this.channels} values, got ${value.length}`);
+        }
+
+        // Calculate the starting point for that pixel, by first determining
+        // what row to start on, then adding the x offset. Finally, take into
+        // account the number of channels.
+        const baseIndex = ((y * this.width) + x) * this.channels;
+
+        // Iterate for each channel, assigning the value to the corresponding
+        // spot in `this.data`.
+        for (let channel = 0; channel < this.channels; channel++) {
+            const channelIndex = baseIndex + channel;
+            if (channelIndex >= this.data.length) {
+                throw new Error('Index out of bounds');
+            }
+            this.data[channelIndex] = value[channel];
+        }
+
+        return this;
+    }
+
+    /**
+     * Apply an alpha mask to the image.
+     * @param {RawImage} mask The mask to apply. Values should be between 0 and 255, and be a single channel.
+     * @returns {Promise<RawImage>} The masked image.
+     * @throws {Error} If the mask is not the same size as the image.
+     * @throws {Error} If the image does not have 4 channels.
+     * @throws {Error} If the mask is not a single channel.
+     */
+    async applyMask(mask) {
+        if (mask.width !== this.width || mask.height !== this.height) {
+            throw new Error('Mask must be the same size as the image');
+        }
+
+        // We want the current image to have an alpha channel, but the mask will
+        // just be a single channel.
+        if (this.channels !== 4) {
+            throw new Error('Image must have 4 channels');
+        }
```

Comment:
Would be useful to support both 3- and 4-channel images as the input image.

Reply:
Would you consider something like this?  
Automatically convert to 4 channels if 3 are provided, however anything else will throw an error.
```suggestion
        if (this.channels === 3) {
            this.convert(4);
        } else if (![3, 4].includes(this.channels)) {
            throw new Error('Image must have 3 or 4 channels');
        }
```

Reply:
To prevent unnecessary conversions, I think we can set the step size accordingly (3 for RGB, 4 for RGBA) and write the image data accordingly.

Reply:
Hey, sorry I’ll get to this soon.
I’m currently poorly and bedridden at the minute, but will work on a fix ASAP.

Reply:
Hey, I’m still not 100% but I’ve been thinking about this.  
How would a 3 channel RGB image work. For setting the alpha? Wouldn’t we always want that fourth alpha channel?